File: position.go
package main

import (
	"fmt"
)

// Функция для ввода числа и позиции
func input() (string, int) {
	var N string
	var P int

	fmt.Print("Введите число N: ")
	fmt.Scan(&N)
	fmt.Print("Введите позицию P: ")
	fmt.Scan(&P)

	return N, P
}

// Функция для вывода символа на указанной позиции
func output(N string, P int) {
	if P <= 0 || P > len(N) {
		fmt.Println("Некорректная позиция P")
	} else {
		result := N[len(N)-P]
		fmt.Println(string(result))
	}
}

func main() {
	N, P := input()
	output(N, P)
}

File: position.cpp
#include <iostream>
#include <string>

using namespace std;

// Функция для ввода числа и позиции
void input(string& N, int& P) {
    cout << "Введите число N: ";
    cin >> N;
    cout << "Введите позицию P: ";
    cin >> P;
}

// Функция для вывода символа на указанной позиции
void output(string& N, int& P) {
    if (P <= 0 || P > N.length()) {// позиция должна быть больше нуля и меньше длинны числа
        cout << "Некорректная позиция P" << endl;
    } else {
        char result = N[N.length() - P];
    cout << result << endl;
    }
}

int main() {
    string N;
    int P;

    input(N, P);
    output(N, P);

    return 0;
}